src/scripts/domParsers/khanAcademyVideoExtractor.ts

```diff
@@ -0,0 +1,50 @@
+import {VideoExtractor} from "./videoExtractor";
+import {VideoUtils} from "./videoUtils";
+import {YoutubeVideoExtractor} from "./youtubeVideoExtractor";
+
+import {Utils} from "../utils";
+
+export class KhanAcademyVideoExtractor implements VideoExtractor {
+	private youtubeExtractor: YoutubeVideoExtractor;
+
+	constructor() {
+		this.youtubeExtractor = new YoutubeVideoExtractor();
+	}
+
+	public getVideoIds(pageUrl: string, pageContent: string): string[] {
+		let regex = /id\s*=\s*("\s*video_(\S+)\s*"|'\s*video_(\S+)\s*')/gi;
```

Comment:
I found this kind of bothersome to do while I was implementing YouTube and Vimeo, but it will probably be useful for the future: could you add comments describing the regexes? :)